src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -607,10 +607,38 @@ void enforceMemberNotNull(SyntaxNode? syntaxOpt, LocalState state)
                         }
                         // https://github.com/dotnet/roslyn/issues/46718: give diagnostics on return points, not constructor signature
                         var exitLocation = method.DeclaringSyntaxReferences.IsEmpty ? null : method.Locations.FirstOrDefault();
+                        var alreadyWarnedMembers = PooledHashSet<Symbol>.GetInstance();
                         foreach (var member in method.ContainingType.GetMembersUnordered())
                         {
-                            checkMemberStateOnConstructorExit(method, member, state, thisSlot, exitLocation);
+                            if (checkMemberStateOnConstructorExit(method, member, state, thisSlot, exitLocation, forcePropertyAnalysis: false) is { } warnedMember)
```

Comment:
It feels like we should be able to determine here whether to give an exit warning for any members of 'ContainingType', and in the second pass only check the base required members.

For example, what if we passed the following argument here (pseudocode): `forcePropertyAnalysis: !currentConstructorEnforcesRequiredMembers && !member.IsAutoProperty() && member.IsRequired`.

It's fine, though, if you found that just tracking the `alreadyWarnedMembers` here ends up being simpler.

Reply:
I hadn't thought of that, thanks!